pkg/errors/errors.go

```diff
@@ -38,9 +62,18 @@ func ValidationErrorsToFieldErrorResponse(errs ValidationErrors) (fieldErrResp F
 		if fieldErr == nil {
 			continue
 		}
+
+		errorType := FieldErrorType
+		if key == "" {
+			// in some rare edge cases we have "validation errors"
+			// that do not have a key value, in this case, we mark them
+			// as generic so that clients do not try to use the empty
+			// key field
+			errorType = GeneralErrorType
+		}
 		fieldErrResp.Errors = append(fieldErrResp.Errors, FieldError{
 			GeneralError: GeneralError{
-				Type:    FieldErrorType,
+				Type:    errorType,
```

Comment:
I would not call it nice but I guess it's okay. Ideally, we would need to switch the `FieldErrorResponse.Errors` field to `[]interface{}` (or some common interface). It does not seem right that we mis-use the `FieldError` struct here.

Reply:
I guess i could update it into an interface, something like `APIError` with a `GetType()` and `GetErrors()`?

Reply:
@LucasRoesler yeah, I think semantically it's important.

Reply:
Done, it had a bigger impact than I expected, but the result is definitely nicer


Here is an example of it in action in my Hub PR 
![image](https://user-images.githubusercontent.com/891889/114750096-e3862d80-9d53-11eb-975c-5a247385518d.png)
in particular, notice that there is no extra `key: ""` anymore because we are not abusing the FieldError anymore :)